src/Android/Services/DeviceActionService.cs

```diff
@@ -267,6 +261,7 @@ bool IsAlive(Java.Lang.Object @object)
             
             var result = new TaskCompletionSource<ValidatablePromptResponse?>();
 
+            alertBuilder.SetOnCancelListener(new BasicDialogWithResultCancelListener(result));
```

Comment:
Just curious, why not just use the same `(sender, args) => result.TrySetResult(null)` as the negative listener?  Is it preventing the dialog from dismissing properly?

Reply:
The negative listener is for the "cancel" button of the dialog and the `SetOnCancelListener` is for when the user taps in the background to dismiss the dialog. Currently, there is no `EventHandler` property to use the form `(sender, args) => result.TrySetResult(null)` so that's why I implemented a custom class for the listener.